### DISCUSSION TITLE
Build index via Github Actions

### DISCUSSION THREAD

**Discussion by tenmozes** [Pagefind/pagefind#477] [Q&A] [answered]
Is there any recommendation on how to build an index via GitHub actions? 

the case - public documentation on GitHub anyone may change, after changes there is deployment on site, and I want to update the index

**Comment by bglw** [comment, marked as answer]
Hey @tenmozes 👋 

I don't know that I have any specific recommendations — generally just running Pagefind in between your build step and your deployment step is all that is needed.

Browsing examples of usage is usually your best bet, to see what others are doing — here's a pretty good GitHub code search for finding existing Pagefind actions:

https://github.com/search?q=pagefind+path%3A.github%2Fworkflows%2F*+-org%3Acloudcannon+&type=code

Let me know if that helps at all, or I'm happy to help dig into to anything more specific 🙂

**Comment by tenmozes** [comment]
@bglw thank you for the link, it covers my question

the main reason why i asked that, i guess the node dependency and npx is a bit overkill for indexing as far pagefind is a compilable binary 
if there is no lightweight solution, i will definitely add node as dependency in github action 

thank you again

**Reply by bglw** [reply]
Oh I get ya — [supplypike/setup-bin](https://github.com/supplypike/setup-bin) is a lovely action for this. I use this for a crate packaged the same way as Pagefind. e.g. in one of Pagefind's actions:

https://github.com/CloudCannon/pagefind/blob/ca82a78e752f8199ebaff6c9aace5fcf812766c0/.github/workflows/test.yml#L78-L83

If you put a Pagefind release `tar.gz` link in there and change the name and version, that action handles downloading, extracting, and putting the binary on the PATH.

**Reply by tenmozes** [reply]
perfect, thank you!
